dirsrvtests/tests/suites/plugins/attruniq_test.py

```diff
@@ -386,3 +553,505 @@ def test_matchingrule_attr(topology_st):
+    attruniq.add_unique_subtree(ACTIVE_DN)
+    attruniq.add_unique_subtree(STAGE_DN)
+    attruniq.enable()
+    topology_st.standalone.restart()
+
+    log.info("Verify both subtrees apply attruniqueness separately")
+    with pytest.raises(ldap.CONSTRAINT_VIOLATION):
+        create_active_user(topology_st, uid=ACTIVE_USER_2_CN, cn=[ACTIVE_USER_1_CN, ACTIVE_USER_2_CN])
+    with pytest.raises(ldap.CONSTRAINT_VIOLATION):
+        create_stage_user(topology_st, uid=STAGE_USER_2_CN, cn=[STAGE_USER_1_CN, STAGE_USER_2_CN])
+
+    log.info("Verify that the subtrees do not enforce uniqueness across each other")
+    try:
+        active_user_2 = create_active_user(topology_st, uid=ACTIVE_USER_2_CN,
+                                           cn=[STAGE_USER_1_CN, ACTIVE_USER_2_CN])
+        stage_user_2 = create_stage_user(topology_st, uid=STAGE_USER_2_CN,
+                                         cn=[ACTIVE_USER_1_CN, STAGE_USER_2_CN])
+    except:
+        assert False, "Failed to create users necessary for the test"
+    finally:
+        active_user_2.delete()
+        stage_user_2.delete()
+
+
+def test_multiple_containers_add_across_subtrees(topology_st, attruniq, containers,
+                           active_user_1, stage_user_1):
+    """Test ADD operations with attribute uniqueness across multiple containers with cross-subtree enforcement.
+
+    This test verifies that when the 'across-all-subtrees' option is enabled,
+    the attribute uniqueness plugin enforces uniqueness across all configured
+    containers, preventing duplicate values anywhere in the configured scope.
+
+    :id: 7b58c46a-9ef1-4bbc-929d-578dbd3f4910
+    :setup: Standalone instance with containers and users in different containers
+    :steps:
+        1. Configure plugin for 'cn' attribute in both containers with across-subtrees enabled
+        2. Try to add user with duplicate 'cn' in active container (should fail)
+        3. Try to add user with duplicate 'cn' in stage container (should fail)
+        4. Try to add user in stage container with 'cn' matching active user (should fail)
+        5. Try to add user in active container with 'cn' matching stage user (should fail)
+    :expectedresults:
+        1. Plugin configured successfully with across-subtrees enabled
+        2. Duplicate in active container fails with CONSTRAINT_VIOLATION
+        3. Duplicate in stage container fails with CONSTRAINT_VIOLATION
+        4. Cross-container duplicate in stage fails with CONSTRAINT_VIOLATION
+        5. Cross-container duplicate in active fails with CONSTRAINT_VIOLATION
+    """
+
+    log.info('Setup attribute uniqueness plugin for "cn" attribute')
+    attruniq.add_unique_subtree(ACTIVE_DN)
+    attruniq.add_unique_subtree(STAGE_DN)
+    attruniq.enable_all_subtrees()
+    attruniq.enable()
+    topology_st.standalone.restart()
+
+    # Inserting sleep to prevent repeatedly occuring 'Server is busy' errors
+    sleep(5)
```

Comment:
@progier389 Pierre, there is an issue with this test, I was advised to turn to you for help, since it needs deeper investigation. The test fails when run together with all the other tests in the file with BDB database with error 51. If the test is run separately it does not encounter the error. The `sleep(5)` prevents the error, but we're not sure what exactly causes the issue.